### PR TITLE
Added completed README.md file and .png file of the circuits.

### PR DESCRIPTION
**Pull request by KleerK**

### CODE REVIEW

**Comment by KleerK** [comment]
By Adrian Tofan.

File: README.md [hunk 1]
```diff
@@ -1,44 +1,128 @@
...
-### c
-
-### d
-
+# COMP110 Worksheet 4
+
+Please edit this README.md file with your answers to the worksheet questions.
+
+I did all of them on paper, then copied them using Markdown. The answers are right, I just hope the starting values have been copied properly from paper.
+## Question 1
```
Comment (edpowley) [inline README.md:6]: ✔

File: README.md [hunk 2]
```diff
@@ -1,44 +1,128 @@
...
+|0|1|0|1|0|
+|0|1|0|0|0|
+|0|0|1|1|0|
+|0|0|1|0|0|
+|0|0|0|1|0|
+|0|0|0|0|0|
+
+## Question 2
+
+![Circuits](CIRCUIT.png)
```
Comment (edpowley) [inline README.md:64]: Missing a `not` gate after the `or` in part d, otherwise ✔

File: README.md [hunk 3]
```diff
@@ -1,44 +1,128 @@
...
+
+## Question 3
+
+### a
+|A|B|A+B|(A+B)'|A'|B'|A' x B'|
+|---|---|---|---|---|---|---|
+|0|1|1|*0*|1|0|*0*|
+|1|0|1|*0*|0|1|*0*|
+|1|1|1|*0*|0|0|*0*|
+|0|0|0|*1*|1|1|*0*|
```
Comment (edpowley) [inline README.md:74]: `A' x B'` should be 1 in the last row?

File: README.md [hunk 4]
```diff
@@ -1,44 +1,128 @@
...
+|1|1|1|*0*|0|0|*0*|
+|0|0|0|*1*|1|1|*1*|
+### c
+|A|B|C|AxB|AxC|(AxB) + (AxC)|A x (B+C)|
+|---|---|---|---|---|---|---|
+|0|0|0|0|0|0|0|
+|0|0|1|0|0|0|0|
+|0|1|0|0|0|0|0|
+|1|0|0|0|0|0|0|
+|0|1|1|0|0|1|1|
```
Comment (edpowley) [inline README.md:89]: Both expressions should come out to 0 in this row

File: README.md [hunk 5]
```diff
@@ -1,44 +1,128 @@
...
+|1|0|1|*0*|0|1|*0*|
+|1|1|1|*0*|0|0|*0*|
+|0|0|0|*1*|1|1|*0*|
+### b
+|A|B|AxB|(AxB)'|A'|B'|A' + B'|
+|---|---|---|---|---|---|---|
+|0|1|0|*1*|1|0|*1*|
+|1|0|0|*1*|0|1|*1*|
+|1|1|1|*0*|0|0|*0*|
+|0|0|0|*1*|1|1|*1*|
```
Comment (edpowley) [inline README.md:81]: ✔

File: README.md [hunk 6]
```diff
@@ -1,44 +1,128 @@
...
+|A|B|C|A+B|A+C|(A+B) x (A+C)|A + (BxC)|
+|---|---|---|---|---|---|---|
+|0|0|0|0|0|0|0|
+|0|0|1|0|1|0|0|
+|0|1|0|1|0|0|0|
+|1|0|0|1|1|1|1|
+|0|1|1|1|1|1|1|
+|1|1|0|1|1|1|1|
+|1|1|1|1|1|1|1|
+|1|0|1|1|1|1|1|
```
Comment (edpowley) [inline README.md:103]: ✔

File: README.md [hunk 7]
```diff
@@ -1,44 +1,128 @@
...
+## Question 4
+
+### a
+The question translates to (AxB)' = A' + B', which works for every possible combination of variable values.
+For the first option, if we do not have both the "a.txt" file and "b.txt" file, then at least one of them is missing. For the second one, if either one of them is missing then at least one is missing.
+### b
+The question translates to (A x B) + (C x A) =  (B + C) x A, which works for every possible combination of variable values.
+For the first option, _Hello_ will be printed if _x_ is either greater than 7 and an integer or greater than 7 and a float. In the second option, if _x_ is either an integer or a float and a number greater than 7, _hello_ is printed on the screen.
+### c
+The question translates to (AxB)' = A' + B'. Comparing the two _if_ statements, we are able to notice the fact that their results are inverted/flipped, the first one masking the fact that it is actually NOT(A and B). The results of the if and else statements are flipped when compared to the other option, underlining the fact that it is a NOT in the first _if_ statement.
```
Comment (edpowley) [inline README.md:114]: ✔

File: README.md [hunk 8]
```diff
@@ -1,44 +1,128 @@
...
+### d
+The question translates to A + (B x C) = B x (A + C), which are *not* equal.
```
Comment (edpowley) [inline README.md:116]: True -- however the cases where the expressions differ correspond to cases where `x > 0` is false but `x > 10` is true, which is clearly impossible. Therefore the two are equal in all cases that are actually possible.

**Pull request closed without merging**